src/Cache/CacheEngineInterface.php

```diff
@@ -25,6 +25,16 @@
  */
 interface CacheEngineInterface
 {
+    /**
+     * Write data for key into a cache engine if it doesn't exist already.
+     *
+     * @param string $key Identifier for the data.
+     * @param mixed $value Data to be cached - anything except a resource.
+     * @return bool True if the data was successfully cached, false on failure.
+     *   Or if the key existed already.
+     */
+    public function add($key, $value);
```

Comment:
I wonder how many users actually need this method? For the few who might be actually using it what this method does could be easily simulated through app code. I would prefer removing it in 4.x in which case we shouldn't add it to the interface.

Reply:
It is only implemented in a few cache engines (memcached, redis, apc). In other engines it is a non-atomic operation.

Reply:
Oh, I only checked the code of `CacheEngine::add()`, if memcache, redis, apc do provide it as atomic operation then I am fine keeping it.

Reply:
Ok. Is there anything else that should change with this?

Reply:
Nope